### ISSUE TITLE
Add auto-detection of container information to resource if cgroupv2 is used

### ISSUE THREAD

**Issue by svrnm** [open-telemetry/opentelemetry-java-instrumentation#6694]
Follow-up to open-telemetry/opentelemetry-java#3308 & related ticket with js: https://github.com/open-telemetry/opentelemetry-js-contrib/issues/1173 

**Is your feature request related to a problem? Please describe.**

Detection of the `container.id` in `ContainerResource.java` currently works by reading from `/proc/self/cgroup`. That does not work when cgroupv2 is turned on. However, based on these two sources there's another place to read from:

- [How to get docker container ID from within the container with cgroup v2](https://stackoverflow.com/questions/68816329/how-to-get-docker-container-id-from-within-the-container-with-cgroup-v2)
- [Python NullResource error when running torizoncore-builder build](https://community.toradex.com/t/python-nullresource-error-when-running-torizoncore-builder-build/15240/4)

In short, the `container.id` could be read from /proc/self/mountinfo as well. This also works for `containerd` and not only docker.

Eventually it would be good to have a standard for that, but unfortunately this is a long-standing non-moving [OCI issue](https://github.com/opencontainers/runtime-spec/issues/1105), until then a combination of both approaches seems to be reliable enough.

**Describe the solution you'd like**
Add code that reads the container id from `/proc/self/mountinfo` if reading from `/proc/self/cgroup` failed.

**Describe alternatives you've considered**
As stated in the description, eventually it would be great to have this standardized, but until then this is the best way to go.

cc: @lo-jason , @PeterF778

**Comment by svrnm** [comment]
@mateuszrzeszutek the code that needs to be changed for that is living in `opentelemetry-java`

https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/resources/src/main/java/io/opentelemetry/sdk/extension/resources/ContainerResource.java

**Comment by svrnm** [comment]
Ah, ok, that's good to know!

**Comment by svrnm** [comment]
thanks @breedx-splk for fixing this :-)

**Comment by Sprakhar97** [comment]
@svrnm I wanted to highlight that this approach will not work for k8s as the container Id in k8s pod spec is different from the one we are extracting from proc/self/mouninfo and hence any correlation with k8s pod will break.

**Comment by breedx-splk** [comment]
> k8s pod spec

@Sprakhar97 Can you link to the part of the spec that you're referring to? Are you saying that the parsing will still work, but the ID is semantically something different?

Any thoughts on how to fix this? Do you know of a better mechanism for finding the current pod id in a compatible (correlatable) way?

**Comment by svrnm** [comment]
I assume this comment from @biswajit-nanda on https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/8462 is related.

@XSAM reported the same issue a while back for go, this is not unique to java.

Overall, there is a major issue with container id detection, since none of the approaches taken is reliable, see also https://github.com/containerd/containerd/issues/8185

**Comment by XSAM** [comment]
> this is not unique to java.

Yes, this could happen to every language's implementation, as the id in the cgroup v2 file may not be the current container's id, it could be the k8s busybox container's id.
